src/sagemaker/workflow/_utils.py

```diff
@@ -493,8 +495,9 @@ def arguments(self) -> RequestType:
             request_dict.pop("Description")
             logger.warning(warn_msg_template, "Description")
         if "ModelPackageName" in request_dict:
-            request_dict.pop("ModelPackageName")
-            logger.warning(warn_msg_template, "ModelPackageName")
+            if not _pipeline_config or not _pipeline_config.use_custom_job_prefix:
```

Comment:
So this is saying even when pipeline config is populated (which will always happen in a PipelineSession) and if there's a prefix flag set, we'll still pop the job name... is that right?

Reply:
Yes, if the config has not been set (which it shouldn't be since we are default initializing it with False in the `Pipeline()`) Constructor OR if it has been set and is toggled off -> false, continue to pop from request_dict as we have been doing.